awx/main/tests/unit/api/test_views.py

```diff
@@ -413,3 +413,17 @@ def test_survey_spec_element_missing_property(self):
         spec['spec'][0].pop('type')
         r = JobTemplateSurveySpec._validate_spec_data(spec, {})
         assert "'type' missing from survey question 0" in r.data['error']
+
+
+    def test_survey_spec_element_integer_empty_default(self):
```

Comment:
These tests are nearly the same.  One pattern you'll see throughout our test suite is to _parametrize_ these with *one* test using `py.test` e.g.,

```python
@pytest.mark.parametrize('_type', ['int', 'float'])
def test_survey_spec_element_number_empty_default(self, _type):
   """ Assert that empty default is allowed for answer. """
    spec = self.spec_from_element({'type': _type, 'default': ''})
    r = JobTemplateSurveySpec._validate_spec_data(spec, {})
    assert r is None

Reply:
I was wondering if there was anything like this, because I didn't want to push the redundant code. Finally got it!